exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/writer/AbstractFixedWidthWriter.java

```diff
@@ -93,17 +112,62 @@ protected final int prepareWrite(int writeIndex) {
     @Override
     protected final void fillEmpties(final int writeIndex) {
       final int width = width();
-      final int stride = ZERO_BUF.length / width;
+      final int stride = emptyValue.length / width;
       int dest = lastWriteIndex + 1;
       while (dest < writeIndex) {
         int length = writeIndex - dest;
         length = Math.min(length, stride);
-        drillBuf.setBytes(dest * width, ZERO_BUF, 0, length * width);
+        drillBuf.setBytes(dest * width, emptyValue, 0, length * width);
         dest += length;
       }
     }
   }
 
+  /**
+   * Base class for writers that use the Java int type as their native
+   * type. Handles common implicit conversions from other types to int.
+   */
+  public static abstract class BaseIntWriter extends BaseFixedWidthWriter {
+
+    @Override
+    public final void setLong(final long value) {
+      try {
+        // Catches int overflow. Does not catch overflow for smaller types.
+        setInt(Math.toIntExact(value));
+      } catch (final ArithmeticException e) {
+        throw InvalidConversionError.writeError(schema(), value, e);
+      }
+    }
+
+    @Override
+    public final void setDouble(final double value) {
```

Comment:
Double covers Float as well?

Reply:
Yes, just as setInt() covers TInyInt, SmallInt, Int, UInt1, and UInt2.